pkg/serviceaccount/openidmetadata.go

```diff
@@ -126,20 +147,22 @@ func NewOpenIDMetadata(issuerURL, jwksURI, defaultExternalAddress string, pubKey
 		}
 	}
 
+	// TODO(liggitt): switch this to an implementation that synchronously constructs at startup (to ensure a successful configuration at startup),
+	// and refreshes configJSON / keysetJSON in the background based on GetCacheMaxAgeSeconds()
+	pubKeys := pubKeyGetter.GetPublicKeys("")
+	if len(pubKeys) == 0 {
+		return nil, fmt.Errorf("no keys provided for validating keyset")
+	}
 	configJSON, err := openIDConfigJSON(issuerURL, jwksURI, pubKeys)
 	if err != nil {
 		return nil, fmt.Errorf("could not marshal issuer discovery JSON, error: %v", err)
 	}
-
 	keysetJSON, err := openIDKeysetJSON(pubKeys)
 	if err != nil {
 		return nil, fmt.Errorf("could not marshal issuer keys JSON, error: %v", err)
 	}
 
-	return &OpenIDMetadata{
-		ConfigJSON:       configJSON,
-		PublicKeysetJSON: keysetJSON,
-	}, nil
+	return NewStaticOpenIDMetadataProvider(3600, configJSON, keysetJSON), nil
```

Comment:
can we take TTL as an arg in NewOpenIDMetadata(...) ?

Reply:
reworked this to make the metadata provider recompute whenever it is told by the public key getter that keys changed